File: src/lib/commerce/sellerPlatform.ts
// Maps a free-text merchant / seller string (Serper `source`, e.g. "Shopee",
// "TikTok Shop", "CellphoneS") to a SMALL FIXED enum for analytics.
//
// 🔒 The raw seller string is NEVER emitted to analytics — only this low-cardinality
// bucket is. It backs the `platform` param of `shopping_search_click`, so a demand
// signal can be sliced by marketplace without leaking the merchant name, the product
// or the URL.

export type ShoppingPlatform = 'shopee' | 'lazada' | 'tiki' | 'tiktok' | 'other'

export function sellerPlatform(seller: string | null | undefined): ShoppingPlatform {
  const s = (seller ?? '').toLowerCase()
  if (s.includes('shopee')) return 'shopee'
  if (s.includes('lazada')) return 'lazada'
  if (s.includes('tiktok')) return 'tiktok' // before 'tiki' so "TikTok" is not mis-bucketed
  if (s.includes('tiki')) return 'tiki'
  return 'other'
}

File: android/app/src/main/java/com/tappyai/app/chat/SellerPlatform.kt
package com.tappyai.app.chat

/**
 * Maps a free-text merchant / seller string (Serper `source`, e.g. "Shopee",
 * "TikTok Shop", "CellphoneS") to a SMALL FIXED enum for analytics — the Kotlin twin
 * of web's `src/lib/commerce/sellerPlatform.ts`.
 *
 * 🔒 The raw seller string is NEVER emitted to analytics — only this low-cardinality
 * bucket is (the `platform` param of `shopping_search_click`).
 */
fun sellerPlatform(seller: String?): String {
    val s = (seller ?: "").lowercase()
    return when {
        s.contains("shopee") -> "shopee"
        s.contains("lazada") -> "lazada"
        s.contains("tiktok") -> "tiktok" // before "tiki" so "TikTok" is not mis-bucketed
        s.contains("tiki") -> "tiki"
        else -> "other"
    }
}
